cartridge/clusterwide-config.lua

```diff
@@ -249,6 +250,37 @@ clusterwide_config_mt = {
             return self._checksum
         end,
 
+        new = function(self, patch)
+            checks('ClusterwideConfig', 'table')
+
+            local new_config = self:copy()
+            for k, v in pairs(patch) do
+                if patch[k] ~= nil and patch[k .. '.yml'] ~= nil then
+                    local err = NewConfigError:new(
+                        'Ambiguous sections %q and %q',
+                        k, k .. '.yml'
+                    )
+                    log.error('%s', err)
```

Comment:
I think it could be moved out of this function. Such simple function shouldn't produce any side-effects.

Reply:
Agreed!